src/event.rs

```diff
@@ -338,9 +339,10 @@ where
 							amount_msat,
 						);
 						self.channel_manager.fail_htlc_backwards(&payment_hash);
-						self.payment_store
-							.update(&payment_hash, None, None, None, Some(PaymentStatus::Failed))
-							.expect("Failed to access payment store");
+
+						let mut update = PaymentDetailsUpdate::new(payment_hash);
+						update.status = Some(PaymentStatus::Failed);
```

Comment:
These may be cleaner using the following notation, especially as the number of fields to set grows. Your call.

```rust
let updata = PaymentDetailsUpdate {
    status: Some(PaymentStatus::Failed),
    ..PaymentDetailsUpdate::new(payment_hash)
};